extra/mariabackup/xtrabackup.cc

```diff
@@ -5100,6 +5181,10 @@ class BackupStages {
 				return false;
 			}
 
+			if (!backup_datasinks.backup_low()) {
+				return false;
+			}
+
 			// Copy log tables tail
 			if (!m_common_backup.copy_log_tables(true)) {
 				msg("Error on copy log tables");
```

Comment:
If `mysql.general_log` is in `ENGINE=CSV`, then the `SHOW ENGINE INNODB STATUS` statement that we must execute somewhere inside `BACKUP STAGE BLOCK_COMMIT` would mark `mysql/general_log.CSM` as dirty and therefore cause a failure of the test `mariabackup.log_tables`. That is why we must first invoke `copy_log_tables` and then ensure that the InnoDB log sequence number has advanced enough.

This piece of information was found by analyzing an `rr replay` trace of the `mariadbd` process during the `mariadb-backup --backup` and after the `mariadb-backup --prepare`.